File: php/features.php
<?php
declare(strict_types=1);

// Jsonplaceholder SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class JsonplaceholderFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new JsonplaceholderBaseFeature();
            case "test":
                return new JsonplaceholderTestFeature();
            default:
                return new JsonplaceholderBaseFeature();
        }
    }
}

File: py/features.py
# Jsonplaceholder SDK feature factory

from feature.base_feature import JsonplaceholderBaseFeature
from feature.test_feature import JsonplaceholderTestFeature


def _make_feature(name):
    features = {
        "base": lambda: JsonplaceholderBaseFeature(),
        "test": lambda: JsonplaceholderTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
